### PR TITLE
Added dropdown to analysis page and connected it to run_analysis

### PR DESCRIPTION
**Pull request by adamcantor22**
# Pull Request Template for MMEDS

## What has changed
Analysis page removed entry for access code and replaced it with a dropdown of study names.

## Checklist of pre-requisites
-   [x] Does the code run?  
-   [x] Does the code follow the repository style?  
-   [x] Is the code tested?  

## How to use the feature
Instead of typing in access code, select study from dropdown.

## How the design has changed
Analysis page has borrowed dropdown format from query page. run_analysis now includes a searching of user studies to connect the study name back to the access code, as access code is no longer given.

## Additional notes
None.

### CODE REVIEW

File: mmeds/server.py [hunk 1]
```diff
@@ -855,6 +855,15 @@ def run_analysis(self, access_code, analysis_method, config, runOnNode=None):
             if runOnNode and not check_privileges(self.get_user(), self.testing):
                 raise err.PrivilegeError('Only users with elevated privileges may run analysis directly')
 
+            with Database(testing=self.testing, owner=self.get_user()) as dbase:
```
Comment (DSWallach) [inline mmeds/server.py:858]:
```suggestion
            with Database(testing=self.testing, owner=self.get_user()) as db:
```
generally `dbase` would be a fine variable name but I've been using `db` in this type of code block and I think it's good to be consistent throughout the project

File: mmeds/server.py [hunk 2]
```diff
@@ -855,6 +855,15 @@ def run_analysis(self, access_code, analysis_method, config, runOnNode=None):
...
+                studies = dbase.get_all_user_studies(self.get_user())
```
Comment (DSWallach) [inline mmeds/server.py:859]: There are method in the Database class for getting the study directly by the access code

**Review by DSWallach — CHANGES_REQUESTED** [review]
Two things for now

File: mmeds/server.py (hunk 2 above)
Comment (adamcantor22) [reply mmeds/server.py:859]: I wasn't able to find these, do you know what they're named?

File: mmeds/database/database.py [hunk 3]
```diff
@@ -897,6 +897,16 @@ def clear_mongo_data(self, username):
         for doc in data:
             doc.delete()
 
+    @classmethod
+    def get_study_from_access_code(cls, code):
+        """ For server use """
+        return MMEDSDoc.objects(access_code=code)[0]
```
Comment (DSWallach) [inline mmeds/database/database.py:903]:
```suggestion
        return MMEDSDoc.objects(access_code=code).first()
```
MMEDSDoc.objects() Returns an iterator not a fully instantiated list so `first()` is more correct

File: mmeds/database/database.py [hunk 4]
```diff
@@ -897,6 +897,16 @@ def clear_mongo_data(self, username):
...
+
+    @classmethod
+    def get_access_code_from_study_name(cls, study):
+        """ For server use """
+        return MMEDSDoc.objects(study_name=study)[0].access_code
```
Comment (DSWallach) [inline mmeds/database/database.py:908]:
```suggestion
        return MMEDSDoc.objects(study_name=study).first().access_code
```

**Review by DSWallach — CHANGES_REQUESTED** [review]

**Review by DSWallach — APPROVED** [review]

**Pull request merged**